apps/consoledot-rhosak/src/routes/data-plane/routes/TopicEditPropertiesRoute.tsx

```diff
@@ -1,13 +1,63 @@
 import type { VoidFunctionComponent } from "react";
-import { EditTopicProperties } from "ui";
+import { useCallback } from "react";
 import type { DataPlaneNavigationProps } from "../routesConsts";
 import { useTopicGate } from "../useTopicGate";
+import { useUpdateTopicMutation } from "consoledot-api";
+import type { Topic } from "ui-models/src/models/topic";
+import { useHistory } from "react-router-dom";
+import { useDispatch } from "react-redux";
 import { DataPlaneTopicHeaderConnected } from "./DataPlaneTopicHeaderConnected";
+import { EditTopicProperties } from "ui";
+import { addNotification } from "@redhat-cloud-services/frontend-components-notifications";
 
 export const TopicEditPropertiesRoute: VoidFunctionComponent<
   DataPlaneNavigationProps
 > = ({ instanceDetailsHref, instanceTopicsHref, instancesHref }) => {
-  const { topic } = useTopicGate();
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+  const history = useHistory();
+  const { instance, topic } = useTopicGate();
+
+  const dispatch = useDispatch();
+
+  const updateTopic = useUpdateTopicMutation();
+
+  const onSave = useCallback(
+    (topicData: Topic) => {
+      void updateTopic.mutateAsync({
+        instanceId: instance.id,
+        adminUrl: instance?.adminUrl || "",
```

Comment:
```suggestion
        adminUrl: instance.adminUrl,
```

Add a `if/throw` check before this to make that code typesafe. Please let's stop shutting down Typescript by providing unsane defaults. Passing `""` will not help the software work ok, it will simply make the error more difficult to spot.